### PR TITLE
1433 networkinfo

### LINKED ISSUE #1433: Action to get network info
As an app developer, I want to get the current device connection details, including IP address

https://pub.dev/packages/network_info_plus

### Proposed EDL

```yaml
View:
  onLoad:
    getNetworkInfo:
      onError:
      onCompleted:
         # provide a map of result here
```

### PR DESCRIPTION
**Pull request by kmahmood74**
fixes #1433 

cc @amin-nas

### CODE REVIEW

File: modules/ensemble/lib/action/get_network_info_action.dart [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  EnsembleAction? onLocationDisabled;
+
+  GetNetworkInfoAction({super.initiator,
+    super.inputs,
+    this.onSuccess,
+    this.onError,
+    this.onDenied,
+    this.onLocationDisabled});
+
+  factory GetNetworkInfoAction.from({Invokable? initiator,dynamic payload}) =>
```
Comment (vusters) [inline modules/ensemble/lib/action/get_network_info_action.dart:29]: you don't need this. just the next one.

File: modules/ensemble/lib/action/get_network_info_action.dart [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+      } else
+      if (locationStatus == LocationPermissionStatus.unableToDetermine.name
+          && onError != null) {
+        return ScreenController().executeAction(context, onError!,
+            event: EnsembleEvent(initiator,
+                data: {
+                  'status': locationStatus
+                }));
+      } else {
+        try {
```
Comment (vusters) [inline modules/ensemble/lib/action/get_network_info_action.dart:70]: You want to throw error if onSuccess is not specified? If you want to force it required, throw a LanguageError if not specified instead.

**Review by vusters — COMMENTED** [review]
You also need to add this Action to action_invokable so Javascript will work. See https://github.com/EnsembleUI/ensemble/wiki/Add-New-Action

**Comment by kmahmood74** [comment]
@vusters the PR is in draft mode, not ready to be reviewed

File: modules/ensemble/lib/action/get_network_info_action.dart (hunk 2 above)
Comment (kmahmood74) [reply modules/ensemble/lib/action/get_network_info_action.dart:70]: will do, thanks

File: modules/ensemble/lib/action/get_network_info_action.dart (hunk 1 above)
Comment (kmahmood74) [reply modules/ensemble/lib/action/get_network_info_action.dart:29]: I do need it for javascript calls, when coming from js, it's not yaml it's json

**Comment by kmahmood74** [comment]
> You also need to add this Action to action_invokable so Javascript will work. See https://github.com/EnsembleUI/ensemble/wiki/Add-New-Action

@vusters this action will *not* work in js. We don't have the ability to do callbacks in js. The callbacks have to be js functions and not actions, today we assume that onSuccess etc are all actions. I'll create a separate ticket to add the ability to do callbacks as people need it to invokeAPI as well. On another thought, perhaps we shouldn't as not having that would encourage the use of yaml. Let me know what you think

**Comment by kmahmood74** [comment]
@vusters ready to be reviewed. also please see my comment about js invokation

**Comment by kmahmood74** [comment]
@vusters let me know when you have had a chance to review this. This is required for a customer

**Review by vusters — APPROVED** [review]

**Pull request merged**